### DISCUSSION TITLE
Latest ufs-weather-model fails on compile

### DISCUSSION THREAD

**Discussion by ericaligo-NOAA** [ufs-community/ufs-weather-model#2073] [Q&A]
Cloned the latest code and the compilation fails quickly. Within ufs-weather-model, I entered:
module purge
module use modulefiles
module load ufs_hera.intel
setenv CMAKE_FLAGS "-DAPP=ATM -DCCPP_SUITES=FV3_HRRR -D32BIT=ON"

followed by:
build.sh > & compile.log

The error messages are below and it's unclear what the problem is.

The compilation failed rather quickly with this message:
Found Python: /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.5.0/envs/unified-env-noavx512/install/intel/2021.5.0/python-3.10.8-oqvn6sa/bin/python3.10
-- Compile stochastic_physics with 64-bit precision to match CCPP slow physics.
Calling CCPP code generator (ccpp_prebuild.py) for suites --suites=FV3_HRRR ...
CMake Error at FV3/ccpp/CMakeLists.txt:40 (message):
  An error occured while running ccpp_prebuild.py, check
  /scratch2/NCEPDEV/fv3-cam/save/Eric.Aligo/ufs_diags_thompson/build/FV3/ccpp/ccpp_prebuild.{out,err}

-- Configuring incomplete, errors occurred!
See also "/scratch2/NCEPDEV/fv3-cam/save/Eric.Aligo/ufs_diags_thompson/build/CMakeFiles/CMakeOutput.log".

Traceback (most recent call last):
  File "/scratch2/NCEPDEV/fv3-cam/save/Eric.Aligo/ufs_diags_thompson/FV3/ccpp/framework/scripts/ccpp_prebuild.py", line 813, in <module>
    main()
  File "/scratch2/NCEPDEV/fv3-cam/save/Eric.Aligo/ufs_diags_thompson/FV3/ccpp/framework/scripts/ccpp_prebuild.py", line 736, in main
    (success, metadata_request, arguments_request, dependencies_request, schemes_in_files) = collect_physics_subroutines(config['scheme_files'])
  File "/scratch2/NCEPDEV/fv3-cam/save/Eric.Aligo/ufs_diags_thompson/FV3/ccpp/framework/scripts/ccpp_prebuild.py", line 328, in collect_physics_subroutines
    (metadata, arguments, dependencies) = parse_scheme_tables(scheme_filepath, scheme_filename)
  File "/scratch2/NCEPDEV/fv3-cam/save/Eric.Aligo/ufs_diags_thompson/FV3/ccpp/framework/scripts/metadata_parser.py", line 766, in parse_scheme_tables
    raise Exception("Dependency {} for scheme table {} does not exit".format(dependency, scheme_name))
Exception: Dependency /scratch2/NCEPDEV/fv3-cam/save/Eric.Aligo/ufs_diags_thompson/FV3/ccpp/physics/physics/Interstitials/UFS_SCM_NEPTUNE/../../Radiation/RRTMGP/rte-rrtmgp/extensions/mo_heating_rates.F90 for scheme table GFS_rrtmgp_post does not exit

**Comment by DeniseWorthen** [comment]
@ericaligo-NOAA Is this something that worked very recently? The way I would do the same thing is (in the tests directory):

```
./compile.sh hera '-DAPP=ATM -DCCPP_SUITES=FV3_HRRR -D32BIT=ON' executable.name intel NO NO 2>&1 | tee executable.name.log
```

where I pick a "executable.name" that makes sense to me. You can swap gnu for intel, and the "NO NO" is for "clean before, clean after".

With this method, no loading of modules etc is required.

**Comment by gspetro-NOAA** [comment]
@ericaligo-NOAA Do you need any further assistance, or did Denise's response resolve the issue? (If so, you can click "Mark as answer"!)
